### ISSUE TITLE
Support for Topics

### ISSUE THREAD

**Issue by orgal-everc** [taskforcesh/bullmq-pro-support#66]
Is it possible with current capabilities to create topic like behaviour over BullMQ?
As in the way RabbitMQ exchanges or Kafka topics behave.
If not, is it in the roadmap or is it practical to try and simulate it applicatively (with Pub/Sub maybe?)
Thanks

**Comment by manast** [comment]
Topics has always been an interesting feature for me, however I am not sure how much interest there is for a feature like this. Would you mind to explain me about your specific use-case to help me better understand the requirement set for a feature like this?

**Comment by orgal-everc** [comment]
Hi absolutely. Basically to leverage redis streams and achieve kafka like behavior. The use case for me is creating a high volume event driven system. Actually a data pipeline composed of a cluster of components which each have an output and can consume the outputs of other components as input. Queues don’t work here because only one consumer can get the message. With streams you can have multiple consumer groups all consuming the same stream for different purposes.

Right now bullmq is a strong contender to systems like rabbit or sqs. With streams you can enter the same space with kafka.

Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________

**Comment by manast** [comment]
Yes, I think that what you propose is quite similar to RabbitMQ exchanges or AWS Topics, this should fit well in current BullMQ design.
